Tests/HttpUnitTests/LoopbackServer.cs

```diff
@@ -0,0 +1,327 @@
+            }
+            return null;
+        }
+
+        public static string GetHttpResponse(HttpStatusCode statusCode = HttpStatusCode.OK, string additionalHeaders = null, string content = null) =>
+            $"HTTP/1.1 {(int)statusCode} {GetStatusDescription(statusCode)}\r\n" +
+            $"Date: {DateTime.UtcNow:R}\r\n" +
+            $"Content-Length: {(content == null ? 0 : content.Length)}\r\n" +
+            additionalHeaders +
+            "\r\n" +
+            content;
+
+        public static string GetSingleChunkHttpResponse(HttpStatusCode statusCode = HttpStatusCode.OK, string additionalHeaders = null, string content = null) =>
+            $"HTTP/1.1 {(int)statusCode} {GetStatusDescription(statusCode)}\r\n" +
+            $"Date: {DateTime.UtcNow:R}\r\n" +
+            "Transfer-Encoding: chunked\r\n" +
+            additionalHeaders +
+            "\r\n" +
+            (string.IsNullOrEmpty(content) ? "" :
+                $"{content.Length:X}\r\n" +
+                $"{content}\r\n") +
+            $"0\r\n" +
+            $"\r\n";
+
+
+        public class Options
+        {
+            public IPAddress Address { get; set; } = IPAddress.Loopback;
+            public int ListenBacklog { get; set; } = 1;
+            public bool UseSsl { get; set; } = false;
+            public SslProtocols SslProtocols { get; set; } = SslProtocols.Tls | SslProtocols.Tls11 | SslProtocols.Tls12;
+            public bool WebSocketEndpoint { get; set; } = false;
+            public Func<Stream, Stream> StreamWrapper { get; set; }
+            public string Username { get; set; }
+            public string Domain { get; set; }
+            public string Password { get; set; }
+            public bool IsProxy { get; set; } = false;
+        }
+
+        public sealed class Connection : IDisposable
+        {
+            private Socket _socket;
+            private Stream _stream;
+            private StreamReader _reader;
+            private StreamWriter _writer;
+
+            public Connection(Socket socket, Stream stream)
+            {
+                _socket = socket;
+                _stream = stream;
+
+                _reader = new StreamReader(stream);
+                _writer = new StreamWriter(stream);
+            }
+
+            public Socket Socket => _socket;
+            public Stream Stream => _stream;
+            public StreamReader Reader => _reader;
+            public StreamWriter Writer => _writer;
+
+            public void Dispose()
+            {
+                try
+                {
+                    // Try to shutdown the send side of the socket.
+                    // This seems to help avoid connection reset issues caused by buffered data
+                    // that has not been sent/acked when the graceful shutdown timeout expires.
+                    // This may throw if the socket was already closed, so eat any exception.
+                    //_socket.Shutdown(SocketShutdown.Send);
```

Comment:
obviously, this line won't do anything. Any reason why it has been commented?

Reply:
All this helper class has been abandoned because of lack of network support in WIN32 nanoCLR. Work on this has to be resumed when that is made available. Plus the conversion from code with generics to code without them. 
All these are work in progress, that's why they are not in the Unit Test project.